packages/flutter/test/cupertino/tableview_test.dart

```diff
@@ -0,0 +1,267 @@
+              SizedBox(width: double.infinity, height: 100.0),
+              SizedBox(width: double.infinity, height: 100.0),
+              SizedBox(width: double.infinity, height: 100.0),
+            ],
+          ),
+        ),
+      );
+
+      // We must pump and settle in this case because an extra frame is required
+      // to report back the height of the last cell. Only then will the dividers
+      // be drawn with the appropriate gap height.
+      await tester.pumpAndSettle();
+
+      // Find the dividers RenderObject and retrieve a description of its state.
+      final RenderObject dividersSearchResult = tester.renderObject(find.byType(CupertinoTableViewExtraDividers));
+      assert(dividersSearchResult != null, 'Could not locate the dividers render object.');
+      final CupertinoTableViewExtraDividersRenderObject dividers = dividersSearchResult;
+      final CupertinoTableViewExtraDividersDescription description = dividers.debugDescribe();
+
+      // We expect the dividers to be as tall as the last cell's height.
+      expect(description.dividerGapHeight, 100.0);
+    });
+
+    testWidgets('CupertinoTableView does not draw extra dividers when content exceeds viewport', (WidgetTester tester) async {
+      final ScrollController scrollController = ScrollController();
+
+      await tester.pumpWidget(
+        CupertinoApp(
+          home: CupertinoTableView.plain(
+            scrollController: scrollController,
+            plainChildren: const <Widget>[
+              SizedBox(width: double.infinity, height: 300.0),
+              SizedBox(width: double.infinity, height: 300.0),
+            ],
+          ),
+        ),
+      );
+
+      // Overscroll past the bottom of the list to force the dividers Sliver to
+      // be created. Once it's created we can verify that it is not drawing dividers.
+      scrollController.jumpTo(600.0);
+      await tester.pumpAndSettle();
+
+      // Find the dividers RenderObject and retrieve a description of its state.
+      final RenderViewport renderViewport = tester.renderObject(find.byType(Viewport));
+      final CupertinoTableViewExtraDividersRenderObject extraDividersSliver = renderViewport.lastChild;
+      final CupertinoTableViewExtraDividersDescription description = extraDividersSliver.debugDescribe();
+
+      // We do NOT expect the dividers to draw themselves because the content of
+      // the CupertinoTableView is scrollable (exceeds the viewport).
+      expect(description.isDrawingDividers, false);
+
+      // When the dividers are not painted, the extent of the extra dividers
+      // should be zero.
+      expect(description.extraDividersExtent, 0.0);
+    });
+  });
+
+  group('Grouped CupertinoTableView', () {
+    // TODO(mattcarroll): fill out all group related tests
+  });
+}
+
+class _TrackingIndexedWidgetBuilder {
+  _TrackingIndexedWidgetBuilder({
+    this.builder,
+  });
+
+  final Set<int> builtIndices = Set<int>();
```

Comment:
Would be interesting for this to be a List and check for things being built too many times too.

Reply:
I refactored the code to provide that ability. Now I'm checking for exactly 1 build for each widget in a range.  This also involved adding a cache to CupertinoTableView.  I also changed the last cell height reporting to use a ChangeNotifier that can now be used by the RenderObject instead of forcing a rebuild.